s3parq/session_helper.py

```diff
@@ -0,0 +1,104 @@
+
+import boto3
+import urllib
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from contextlib import contextmanager
+import logging
+
+
+success = True
+fail = False
+
+
+class SessionHelper:  
+    '''
+    Helper class for establishing a connection with a Redshift database.  It assumes your IAM
+    role has been configured to access the cluster _and_ has privileges on the database.
+
+    Example usage:
+    SH = SessionHelper('us-east-1', 'some-cluster-1', 'some-host','5439','some-database')
+    SH.configure_session_helper()
+    '''
+
+    def __init__(self, region: str, cluster_id: str, host: str, port: str, db_name: str):
+        self.region = region
+        self.cluster_id = cluster_id
+        self.host = host
+        self.port = port
+        self.db_name = db_name
+
+    def set_boto_session(self):
+        try:
+            self.boto_session = boto3.Session(region_name=self.region)
+            return success
+        except Exception as e:  # TODO: which specific errors can occur?
```

Comment:
What do we gain by doing the global try excepts here and in other setup functions?

Reply:
We gain the ability to know if it failed/succeeded, since it only assigns local variables and doesn't return anything.  But what I think you're saying is to remove it and only use it in the `configure_session_helper` around everything, that way we'll still be alerted if it doesn't connect properly
----
i'm gonna do that, it should make things cleaner

Reply:
@schafrn just pushed that change

Reply:
Perfect! Thank you. That preserves the original exception + since these are technically public functions keeps their responses clear - those were my main worries depending on what we needed and the new change fits all of it I think 👌